modules/ROOT/pages/http/http-listener-ref.adoc

```diff
@@ -1,182 +1,296 @@
+
+An expression like `#['Received a ' ++ attributes.method ++ ' request for account ' ++ attributes.uriParams.accountId ++ '. The following users will be notified: ' ++ (attributes.queryParams.*notify joinBy ', ')]` will return `Received a POST request for account salesforce. The following users will be notified: admin, jane.doe`.
+
+See the xref:http/http-documentation.adoc[HTTP Documentation Reference] for the complete and detailed listing of HTTP request attributes.
+
+==== Multipart Form Example
+
+The following example shows the manipulation of an HTTP Listener source message when an HTML form is received, such as:
 
 [source,xml,linenums]
 ----
 <form action="http://server.com/cgi/handle"
         enctype="multipart/form-data"
         method="post">
 
-    What is your name? <INPUT type="text" name="name"><BR>
-    What is your request? <INPUT type="file" name="request"><BR>
-    What is your favorite color? <INPUT type="text" name="color"><BR>
+    How would you like to identify the logo? <INPUT type="text" name="name"><BR>
+    Which is the logo file? <INPUT type="file" name="logo"><BR>
+    What is the main color in the logo? <INPUT type="text" name="color"><BR>
     <INPUT type="submit" value="Send"> <INPUT type="reset">
 
 </form>
 ----
 
-The request headers are:
+The resulting HTTP request when submitting the form is a multipart one:
 
 [source,text,linenums]
 ----
-POST /accounts HTTP/1.1
+POST /api/v1/account/mulesoft/logo HTTP/1.1
+Content-Type: multipart/form-data; boundary=489691234097965980223899
 Host: localhost:8081
-Cache-Control: no-cache
-User-Agent: Mule/4.0
-Content-Type: multipart/form-data; boundary=AaB03x
+content-length: 34332
 
- --AaB03x
+--489691234097965980223899
 Content-Disposition: form-data; name="name"
 
-MuleSoft
- --AaB03x
-Content-Disposition: form-data; name="request"; filename="myrequest.png"
+Corporate Logo
+--489691234097965980223899
+Content-Disposition: form-data; name="logo"; filename="MuleSoft_logo.png"
 Content-Type: image/png
 
-
- --AaB03x
+.PNG
+.
+...
+IHDR.......L......~.....	pHYs...#...#.x.?v.. .IDATx....q.W.6.....~".N....t....t..#.....LD0T.CF0b..:.3......Q..@...q]U*y\c....
+....`%.$....V"H....`%.$....V"H....`%.$....V"H....`%.$....V"H....`%.$....V"H....`%.$....V"H....`%.$....V"H....`%.$....V"H....`%
+.$....V"H....`%.$....V"H....`%.$....V"H....`%.$....V"H....`%.$....V"H....`%.$....^6.......|..P.....IEND.B`.
+--489691234097965980223899
 Content-Disposition: form-data; name="color"
 
 blue
- --AaB03x
+--489691234097965980223899--
 ----
 
-== Using DataWeave to Access the Form Data
-
-You access the form values using expressions that resemble these:
+Each item can be accessed through the `parts` object, either by name or item number. For example, the second part can be accessed with `payload.parts.logo` or `payload.parts[1]`. The latter is useful when a name is not provided.
```

Comment:
maybe add examples of:

1) I have N parts, they all specify a json mimeType
2) I have N parts, they all hold json payloads but they don't specify it in the request
3) I have N parts, they are all of different mimeTypes, some specified, some not.

Reply:
I don't understand what the difference is if they specify a mime type or not, all are the same or not, etc. The expressions to access the data are always the same. Or are you saying to explain what a lack of content-type means?